File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/error/GenericException.java
package com.github.dedis.popstellar.utility.error;

import androidx.annotation.StringRes;

public abstract class GenericException extends Exception {

  protected GenericException(String message) {
    super(message);
  }

  protected GenericException(String message, Throwable cause) {
    super(message, cause);
  }

  protected GenericException() {}

  @StringRes
  public abstract int getUserMessage();

  public abstract Object[] getUserMessageArguments();
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/error/GenericException.kt
package com.github.dedis.popstellar.utility.error

import androidx.annotation.StringRes

abstract class GenericException : Exception {
    protected constructor(message: String?) : super(message)
    protected constructor(message: String?, cause: Throwable?) : super(message, cause)
    protected constructor()

    @get:StringRes
    abstract val userMessage: Int
    abstract val userMessageArguments: Array<Any?>
}